[16.0] [MIG] stock_barcodes

FranzPoize opened the pull request.

sergio-teruel committed:
[ADD] stock_barcodes: New module to read barcodes on stocks operations (#195)

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update stock_barcodes.pot

astirpe committed:
[12.0][MIG] stock_barcodes

sergio-teruel committed:
[ADD] stock_barcodes: Allow to read barcodes from pickings

pedrobaeza committed:
[FIX] stock_barcodes: Fix ES translation

sergio-teruel committed:
[ADD] stock_barcodes: Allow to read barcodes from pickings. Last changes

sergio-teruel committed:
[FIX] stock_barcodes: Fix validate picking from candidate picking kanban view

sergio-teruel committed:
[IMP] stock_barcodes: Hidden lot for non tracking products.
When user creates a lot, set default product the wizard selected product.
When confirm the lot creation wizard, set this lot in lot wizard scanning barcode field.

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-11.0/stock-logistics-barcode-11.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-11-0/stock-logistics-barcode-11-0-stock_barcodes/

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-12.0/stock-logistics-barcode-12.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-12-0/stock-logistics-barcode-12-0-stock_barcodes/

chienandalu committed:
[FIX] stock_barcodes: out picking location workflow

When we've got an out picking, the logical workflow is that the scanned location
is the location we're getting the stock from

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 12.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-12.0/stock-logistics-barcode-12.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-12-0/stock-logistics-barcode-12-0-stock_barcodes/

chienandalu committed:
[FIX] stock_barcodes: location_dest_id

OCA-git-bot committed:
stock_barcodes 12.0.1.1.1

sergio-teruel committed:
[IMP] stock_barcodes: black, isort

sergio-teruel committed:
[MIG] stock_barcodes: Migration to v13.0

oca-travis committed:
[UPD] Update stock_barcodes.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-13.0/stock-logistics-barcode-13.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-13-0/stock-logistics-barcode-13-0-stock_barcodes/

pedrobaeza committed:
[FIX] stock_barcodes: Make tests more resilient

UID = 1 is sometimes called "System" and sometimes "OdooBot", depending on the
installed set of modules, so tests can fail if you rely on this data.

Removing that variable from the tests, we get resilient tests.

pedrobaeza committed:
[FIX] stock_barcodes: Another non resilient test

Followup of dab52c9. There's another test that hardcodes OdooBot name for uid=1.

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-13.0/stock-logistics-barcode-13.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-13-0/stock-logistics-barcode-13-0-stock_barcodes/

sergio-teruel committed:
[IMP] stock_barcodes: Add barcode field to location view in same group like as v12.0

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 13.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-13.0/stock-logistics-barcode-13.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-13-0/stock-logistics-barcode-13-0-stock_barcodes/

chienandalu committed:
[IMP] stock_barcodes: allow to add lines even if not qtys reserved

If we had any reservation (i.e unreserve), the barcodes button was
hidden.

sergio-teruel committed:
[ADD] stock_barcodes: Add option to get lots automatically in inventory depends on removal strategy
TT27844

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
stock_barcodes 13.0.1.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-13.0/stock-logistics-barcode-13.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-13-0/stock-logistics-barcode-13-0-stock_barcodes/

fshah-initos committed:
[IMP] stock_barcodes: black, isort, prettier

fshah-initos committed:
[14.0][MIG]stock_barcodes

etobella committed:
[FIX] stock_barcodes: use the uom of the package if defined

etobella committed:
[IMP] stock_barcodes: Apply putaway strategy

chienandalu committed:
[FIX] stock_barcodes: avoid mixing tracked and untracked quants

Using the auto lot feature in the inventory scanning we want to avoid
distributing quantities to quants with no lot if the products if the
product change from untracked to tracked.

TT31267

chienandalu committed:
[FIX] stock_barcodes: reset lot id if we change the product manually

Otherwise the previous product lot would persist

TT31267

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

sergio-teruel committed:
[IMP] stock_barcodes: Refactoring code and improvements

OCA-git-bot committed:
stock_barcodes 14.0.1.1.0

mariadforgeflow committed:
[FIX] stock_barcodes: make available the action to read barcodes

OCA-git-bot committed:
stock_barcodes 14.0.1.2.0

LoisRForgeFlow committed:
[FIX] stock_barcodes: take company from env not from user

OCA-git-bot committed:
stock_barcodes 14.0.1.2.1

LoisRForgeFlow committed:
[FIX] stock_barcodes: read act windows permission issue

OCA-git-bot committed:
stock_barcodes 14.0.1.2.2

LoisRForgeFlow committed:
[FIX] stock_barcodes: set option to create lot to true in test

It was de default behavior in GS1 before the refactor

LoisRForgeFlow committed:
[FIX] stock_barcodes: several small adjustments.

* When a picking is processed partially it was not
  posible to validate it because the backorder wizard
  was captured and never returned to the user.
* Only update sml locations when there is a location
  to update, otherwise constraints for non null values
  could raise.
* Invalidate cache of todo lines when determining
  next action to do to ensure that line status is
  updated.
* Reword string in button from "Force" to "Done"
  as it was a bit misleading. Actually, in spanish
  translation it was already being translated to
  "Hecho" ("Done").

LoisRForgeFlow committed:
[IMP] stock_barcodes: show quantity even when not editable.

LoisRForgeFlow committed:
[MIG] stock_barcodes: clean warnings and adapt to v14.

Adapt last refector done in v13 to v14.

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
stock_barcodes 14.0.2.0.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/

francesco-ooops committed:
Added translation using Weblate (Italian)

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 35.6% (94 of 264 strings)

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 35.9% (95 of 264 strings)

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (264 of 264 strings)

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/it/

LoisRForgeFlow committed:
[FIX] stock_barcodes: make actions readable by non-admin users.

leftover from migration.

LoisRForgeFlow committed:
[FIX] stock_barcodes: do not use PO UoM by default

When creating new stock move lines, we should use the UoM in the
move, if any, or the default UoM of the product.

OCA-git-bot committed:
stock_barcodes 14.0.2.0.1

LoisRForgeFlow committed:
[FIX] stock_barcodes: remove last scan do not revert last action.

In pickings, the scan log was always generated without details of
the move lines modified/created. Without this information in the log
removing the log was dong nothing else than just that, removing the
log; leaving the last operation still reflected in the picking.

Also, when the picking was a draft one, the stock move line and
the stock move could have been created by the scan itself, so we
need to remove it in that case, reverting the action would leave
the stock move there that could be validated by mistake.

LoisRForgeFlow committed:
[FIX] stock_barcodes: use float_compare to avoid rounding issues

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 14.0.2.0.2

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/

sergio-teruel committed:
[IMP] stock_barcodes: Clean wizard values after the move to do has been reset

OCA-git-bot committed:
stock_barcodes 14.0.2.0.3

etobella committed:
[IMP] stock_barcodes: black, isort, prettier

etobella committed:
[MIG] stock_barcodes: Migration to 15.0

mariadforgeflow committed:
[IMP] stock_barcodes: add Barcode Option Group for not planned pickings

LoisRForgeFlow committed:
[MIG/FIX] stock_barcodes: fix notification usage to adapt to v15.

LoisRForgeFlow committed:
[IMP] stock_barcodes: attend new pylint mandatory checks

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 15.0.1.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

MiquelRForgeFlow committed:
[FIX] stock_barcodes: add migration script

sergio-teruel committed:
[FIX] stock_barcodes: Do not create a lot with create lot option enabled for products without tracking

sergio-teruel committed:
[FIX] stock_barcodes: Pending moves are not displayed when moves origin is stock_moves without reservations

OCA-git-bot committed:
stock_barcodes 15.0.1.2.0

sergio-teruel committed:
[FIX] stock_barcodes: put in pack call method fails

OCA-git-bot committed:
stock_barcodes 15.0.1.2.1

sergio-teruel committed:
[IMP] stock_barcodes: take into account package fields when scan barcodes

sergio-teruel committed:
[FIX] stock_barcodes: Not pending moves when origin pending moves is stock move lines
TT41789

OCA-git-bot committed:
stock_barcodes 15.0.1.2.2

sergio-teruel committed:
[IMP] stock_barcodes: Add prepare method to set wizard values from picking to be more inheritable

pere-aquarian committed:
Translated using Weblate (Spanish)

Currently translated at 69.1% (175 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Allow display detailed operations in barcode wizard

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Display partner on wizard

sergio-teruel committed:
[IMP] stock_barcodes: Do not clean values forced in an action done

sergio-teruel committed:
[IMP] stock_barcodes: add package to moves todo group key

sergio-teruel committed:
[IMP] stock_barcodes: display picking notes in wizard barcode

sergio-teruel committed:
[IMP] stock_barcodes: display improvements

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

carlosdauden committed:
[IMP] stock_barcodes: Show same type ready pickings tree after validate picking

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Allow make inventories

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[FIX] stock_barcodes: Check serial product tracking for inventory when a quant is created

sergio-teruel committed:
[IMP] stock_barcodes: Improve heritability of write on stock move lines

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

OCA-git-bot committed:
[UPD] README.rst

sergio-teruel committed:
[IMP] stock_barcodes: Better inheritance of qty focus

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Add hook to process values when moves todo have been created.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Test improvements and create lots in confirm action

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (253 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/es/

Commit:
[UPD] Update stock_barcodes.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (276 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/es/

sergio-teruel committed:
[IMP] stock_barcodes: Filled all fields if they is defined in options as filled default.
Determine scan step after determine todo action

sergio-teruel committed:
[FIX] stock_barcodes: Incorrect order of detailed operations fixed

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] Add security group to barcodes main menu

sergio-teruel committed:
[IMP] stock_barcodes: Allow skip clean values when a sml is created or updated in action_done method.

sergio-teruel committed:
[IMP] stock_barcodes: Filter sml lines bby complete packages

sergio-teruel committed:
[IMP] stock_barcodes: Display only the quants pending to read. Allow set to zero

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Add option to keep wizard values until the pending move is completed

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Try fill wizard data after product is read if fill_fields_from_lot is checked

sergio-teruel committed:
[IMP] stock_barcodes: Add total product qty without lot, package, etc

sergio-teruel committed:
[IMP] stock_barcodes: Set order in inventory quants

CarlosRoca13 committed:
[IMP] stock_barcodes: Visual improvements on footer

sergio-teruel committed:
[FIX] stock_barcodes: Reload todo action after unlink sml done

sergio-teruel committed:
[IMP] stock_barcodes Set to readonly total qty fields

sergio-teruel committed:
[IMP] stock_barcodes: Recompute fill_todo_records after unlink sml.

sergio-teruel committed:
[IMP] stock_barcodes: make todo qty_done compute readonly to better computation of pending lines

CarlosRoca13 committed:
[IMP] stock_barcodes: Responsive dropdown

sergio-teruel committed:
[IMP] stock_barcodes: Return stock move line created or updated after confirm action

carlosdauden committed:
[IMP] stock_barcodes: Ignore rest instead of Done to avoid that user skip pending operations

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Inject a context to know when a picking is validated from barcodes interface

sergio-teruel committed:
[FIX] stock_barcodes: Set incorrect package and owner in set info from quants

sergio-teruel committed:
[FIX] stock_barcodes: create new lines when qty_done is greather than initial demand
TT46503

sergio-teruel committed:
[FIX] stock_barcodes: qty_done is increased if user scan a package twice

sergio-teruel committed:
[FIX] stock_barcodes: Link to correct move whe a sml is created

sergio-teruel committed:
[FIX] stock_barcodes: Apply locations domain in pickings and inventory

sergio-teruel committed:
[FIX] stock_barcodes: Always re-write las location scanned is not candidate sml's was found

sergio-teruel committed:
[IMP] stock_barcodes: Add context to avoid update quantities from lot if fill data from lot is active

sergio-teruel committed:
[IMP] stock_barcodes: Search packages in all internal locations is forced option is not checked

sergio-teruel committed:
[FIX] stock_barcodes: The package is not taken into account when user does an inventory

carlosdauden committed:
[IMP] stock_barcodes: Add accumulate_read_quantity option to be used in inventory. In the future it can be used by other process.
TT46934

carlosdauden committed:
[IMP] stock_barcodes: Allow find package in external locations. Add barcode_domain_field context
TT46823

carlosdauden committed:
[IMP] stock_barcodes: Fill inventory quantity when scan package
TT46934

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

carlosdauden committed:
[IMP] stock_barcodes: Add option ignore_quant_location
TT47469

sergio-teruel committed:
[FIX] stock_barcodes: product not take into account to search quants.

sergio-teruel committed:
[IMP] stock_barcodes: Limit product quantities entry to avoid barcode read errors over an input element
TT48037

FranzPoize committed:
[MIG] stock_barcodes: Migration to 16.0

sergio-teruel committed:
[IMP] stock_barcodes: Allow prefill product_qty when not in manual entry mode
TT48077

angelmoya commented:
Functional testing:

Set Barcode options:
![Odoo-Picking-IN-options](https://github.com/OCA/stock-logistics-barcode/assets/6750341/1c5ad01c-5f5a-4534-9d72-5087a6affd9f)

Set this barcode option on incoming picking type.

Configura one product tracking as serial number.

Purchase this product to create a incoming picking.

Go to incoming picking and get this error trying to open barcode interface:

Traceback (most recent call last):
  File "/opt/odoo/odoo/http.py", line 1633, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/opt/odoo/odoo/service/model.py", line 133, in retrying
    result = func()
  File "/opt/odoo/odoo/http.py", line 1660, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/opt/odoo/odoo/http.py", line 1864, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 154, in _dispatch
    result = endpoint(**request.params)
  File "/opt/odoo/odoo/http.py", line 697, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 46, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 33, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 468, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 453, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/mnt/data/odoo-addons-dir/stock_barcodes/models/stock_picking.py", line 35, in action_barcode_scan
    wiz.determine_todo_action()
  File "/mnt/data/odoo-addons-dir/stock_barcodes/wizard/stock_barcodes_read_picking.py", line 188, in determine_todo_action
    self.fill_todo_records()
  File "/mnt/data/odoo-addons-dir/stock_barcodes/wizard/stock_barcodes_read_picking.py", line 181, in fill_todo_records
    self.env["wiz.stock.barcodes.read.todo"].fill_records(self, [move_lines])
  File "/mnt/data/odoo-addons-dir/stock_barcodes/wizard/stock_barcodes_read_todo.py", line 163, in fill_records
    todo_vals[key] = self._prepare_fill_record_values(
  File "/mnt/data/odoo-addons-dir/stock_barcodes/wizard/stock_barcodes_read_todo.py", line 88, in _prepare_fill_record_values
    "product_uom_qty": line.product_uom_qty,
AttributeError: 'stock.move.line' object has no attribute 'product_uom_qty'

sergio-teruel committed:
[IMP] stock_barcodes: Add context to skip set info from quants although the configuration option is activated

sergio-teruel committed:
[IMP] stock_barcodes: Allow display notifications to web client

sergio-teruel committed:
[IMP] stock_barcodes: Allow apply putaway strategy as screen option

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

chienandalu commented on stock_barcodes/__manifest__.py:
```diff
@@ -3,12 +3,12 @@
     "license": "AGPL-3",
     "category": "Extra Tools",
-    "depends": ["barcodes", "stock", "web_widget_numeric_step"],
+    "depends": ["barcodes", "stock", "web_widget_numeric_step", "web"],
```
`web_widget_numeric_step` already depends on web

chienandalu commented on stock_barcodes/wizard/stock_barcodes_read_picking_views.xml:
```diff
@@ -125,121 +126,116 @@
+                                                </div>
+                                                <div class="col-2">
+                                                    <span
+                                                        class="pull-right text-right text-muted"
```
Review your bootstrap classes as they should be migrated to BS5: https://getbootstrap.com/docs/5.0/migration/

You've got some tooling in `openupgradelib` to analyze your view code: https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade_160.py

chienandalu commented on stock_barcodes/static/src/barcodes_models_utils.esm.js:
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2022 Tecnativa - Alexandre D. Díaz
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+const barcodeModels = [
```
```suggestion
export const barcodeModels = [
```

FranzPoize replied:
Yes thank you !

chienandalu commented on stock_barcodes/static/src/barcodes_models_utils.esm.js:
```diff
@@ -0,0 +1,19 @@
+];
+
+export function isAllowedBarcodeModel(modelName) {
+    return barcodeModels.indexOf(modelName) !== -1;
```
A little bit more expressive:
```suggestion
    return barcodeModels.includes(modelName);

FranzPoize replied:
Thank you ! Changed

chienandalu commented on stock_barcodes/static/src/barcodes_models_utils.esm.js:
Organize these source files at leat in functionality folders so it's more clear where to look what

FranzPoize replied:
Tried to. Tell me if you thinks that's better

chienandalu commented on stock_barcodes/static/src/barcodes_models_utils.esm.js:
```diff
@@ -0,0 +1,19 @@
+    "wiz.stock.barcodes.read.todo",
+];
+
+export function isAllowedBarcodeModel(modelName) {
```
This also had a docstring

FranzPoize replied:
Added

chienandalu commented on stock_barcodes/static/src/boolean_toggle.esm.js:
```diff
@@ -0,0 +1,17 @@
+import {registry} from "@web/core/registry";
+
+class BarcodeBooleanToggleField extends BooleanToggleField {
+    onChange(newValue) {
```
Missing documentation from the original here as well where it was stated why this was needed...

FranzPoize replied:
Added !

chienandalu commented on stock_barcodes/static/src/form_view.esm.js:
```diff
@@ -0,0 +1,15 @@
+import {patch} from "@web/core/utils/patch";
+
+patch(FormController.prototype, "Allow display.controlPanel overriding", {
+    setup() {
```
Docs missing here as well

FranzPoize replied:
Added !

chienandalu commented on stock_barcodes/static/src/kanban_renderer.esm.js:
```diff
@@ -0,0 +1,163 @@
+            .map((group) => [...group.querySelectorAll(".o_kanban_record")])
+            .filter((group) => group.length);
+
+        if (isAllowedBarcodeModel(this.props.list.resModel)) {
```
Why do yo need to copy the whole method if you're returning in the middle of it? Maybe you have to repeat some of it if it's encapsulated, but in the end you should try to return `super`. Just evaluate this first, and repeat as less as possible

FranzPoize replied:
The code is not returning in the `if` statement it's recomputing the `cards` variable.
I could call super and return its value but I would still have duplicate that code because nothing is returned from the parent function. I still need to compute the candidate `cards` still filter them and focus the correct one.

If you have a way to do it better I would need more information because I'm not sure how to go about it.

chienandalu commented on stock_barcodes/static/src/kanban_renderer.esm.js:
```diff
@@ -0,0 +1,163 @@
+    setup() {
+        const rootRef = useRef("root");
+        useHotkey(
+            "Enter",
```
What's the purpose of this?

FranzPoize replied:
Not sure what you're referring to ? the rootRef or the useHotkey ?

chienandalu reviewed (changes requested):
Hi @FranzPoize Thanks for your work here.

I didn't go deep into the python logic, although it looks correct at first sight.

As a general comment, I'm finding hard to review the changes in the javascript logic:

- There are no diffs
- There's missing documentation from the former code and for the new it isn't explained.
- There are overrides that copy the original method (a thus the inheritance is broken for those)

Some more specific comments:

FranzPoize commented:
Thank you @chienandalu for your review

> * There are no diffs

Yes unfortunately the differences in behavior between the legacy front and owl are truly unreconcialable...
First of all the file name are changed to use the new odoo-module.
Also for example most of the logic for hotkey management was in kanban renderer. In the new frontend there can be many kanban renderer instances which would have registered their own global management of hotkey this needed to be completely refactored.
> * There's missing documentation from the former code and for the new it isn't explained.

I hope it's fixed in the latest push
> * There are overrides that copy the original method (a thus the inheritance is broken for those)

Yes that is not inevitable but is the least ugly way to do what needs to be done I'll go in more detail in the review comment

chienandalu reviewed (commented):
Thanks @FranzPoize take also in consideration that there are lots of recent changes in the 15.0 that should be included as a base to the migration as well...

chienandalu reviewed (changes requested):
Also testing a manual entry, I found this error

```
File "/opt/odoo/auto/addons/stock_barcodes/wizard/stock_barcodes_read.py", line 744, in action_confirm
    res = self.action_done()
  File "/opt/odoo/auto/addons/stock_barcodes/wizard/stock_barcodes_read_picking.py", line 269, in action_done
    ).action_done()
  File "/opt/odoo/auto/addons/stock_barcodes/wizard/stock_barcodes_read.py", line 547, in action_done
    if not self.check_done_conditions():
  File "/opt/odoo/auto/addons/stock_barcodes/wizard/stock_barcodes_read_picking.py", line 670, in check_done_conditions
    and float_compare(
  File "/opt/odoo/custom/src/odoo/odoo/tools/float_utils.py", line 155, in float_compare
    rounding_factor = _float_check_precision(precision_digits=precision_digits,
  File "/opt/odoo/custom/src/odoo/odoo/tools/float_utils.py", line 29, in _float_check_precision
    assert precision_rounding is None or precision_rounding > 0,\
AssertionError: precision_rounding must be positive, got 0.0

The above server error caused the following client error:
RPC_ERROR: Odoo Server Error
    at makeErrorFromResponse (http://localhost:16069/web/assets/3930-e51fdb0/web.assets_backend.min.js:993:163)
    at XMLHttpRequest.<anonymous> (http://localhost:16069/web/assets/3930-e51fdb0/web.assets_backend.min.js:1001:13)

FranzPoize commented:
Yes this PR includes all the changes up to 28th of february not sure I'll have the bandwidth to add the more recent changes if somebody can help with that I'd be glad.

I'll fix the codecov issue

sergio-teruel committed:
[FIX] stock_barcodes: Required lot_id although lot_name is set without create lot option is checked.
Display notification for required field.

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

chienandalu commented:
> Yes this PR includes all the changes up to 28th of february not sure I'll have the bandwidth to add the more recent changes if somebody can help with that I'd be glad.
> 
> I'll fix the codecov issue

We can do this: finish your ongoing fixes and then we can supersed the migration from the 15.0 branch and putting on top your migration commit fixing conflicts and finishing whatever needs to be fixed from there. What do you think?

carolinafernandez-tecnativa commented on stock_barcodes/wizard/stock_barcodes_read.py:
```diff
@@ -0,0 +1,745 @@
+        }
+
+    def _create_new_lot(self):
+        StockProductionLot = self.env["stock.production.lot"]
```
Hi, you need to change this model to stock.lot

        StockProductionLot = self.env["stock.lot"]

chienandalu commented:
Hi, @FranzPoize what do you think about what I proposed?:

> We can do this: finish your ongoing fixes and then we can supersed the migration from the 15.0 branch and putting on top your migration commit fixing conflicts and finishing whatever needs to be fixed from there. What do you think?

sergio-teruel committed:
[IMP] stock_barcodes: Add option group to allow to select if you want order by location or location_dest_id

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

FranzPoize committed:
[IMP] stock_barcodes: black, isort, prettier

chienandalu committed:
[MIG] stock_barcodes: followup migration

TT44219

FranzPoize commented:
@chienandalu 
I'm having a real trouble figuring how the code works.

When you use the barcode button on a picking type. It calls `action_barcode_scan` on the `stock_picking_type` models there is 3 methods that are called:
* `create`
* `determine_todo_action`
* `fill_pending_moves`

1. First called is `create` this call `set_candidate_picking` if there is a `picking_id` but there is no `picking_id` in the parameters (even when we click on the new button on a picking type). It then calls `action_show_step` which sets the correct step because most `filled_default` are still correctly set (by `_prepare_barcode_wiz_vals`)

2. Second is `determine_todo_action`, if the barcode option is guided, this calls `fill_todo_records` if there is no `todo_line_ids` (and there isn't) which calls `_get_stock_move_lines`. Since there is no `picking_id`, `_get_stock_move_lines` has to return an empty list. In this method all the default are filled but since there is no `todo_line_ids` all those value are set to empty sets. This then calls `action_show_step` again which remove the properly set step because the default value are now all empty.

3. Third a call to `fill_pending_moves` this call `fill_todo_records` again which still does nothing.

So during `action_barcode_scan`:
* `create` sets the proper default values
* `action_show_step` is called twice once it sets a step (during) `create` and the second time it undoes what it did because `determine_todo_action` unset all the default values
* `fill_todo_records` is called twice and twice it does nothing

This means no step at the end in some case (which are real convoluted but not so much).

* Set the barcode option as guided
* Leave no step with a step number of 0 (if a step with a 0 selected is `action_show_step` it will trigger the line`self.step = options_required[:1].step`because `not 0` is `True`).
* And click on the barcode of the `picking_type`

Or simpler use the demo Picking internals barcode option on internal transfer and click on the barcode on the internal transfer picking type.

rousseldenis commented:
/ocabot migration stock_barcodes

FranzPoize commented:
Found the commit that adds a new call to `action_show_step` in `determine_todo_action` 50b0c67107c2c618ceff60752222305df10659b5 

This seems to break the option/step finding since if there is no `todo_line_ids` ,which there isn't in all case where you click on a barcode button on a picking type.

My question is it suppose to work without a picking_id in the `stock_barcodes_read_picking` ? There is a picking_ids field but it is not used anywhere in the code.

@chienandalu @sergio-teruel

FranzPoize committed:
[MIG] stock_barcodes: more migrating

pedrobaeza commented:
To be taken into account: #605

FranzPoize commented:
The frontend has been completely redone this should be resolved in 16.0

Le jeu. 2 mai 2024 à 12:56, Pedro M. Baeza ***@***.***> a
écrit :

> To be taken into account: #605
> <https://github.com/OCA/stock-logistics-barcode/pull/605>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/stock-logistics-barcode/pull/587#issuecomment-2090201915>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHNHZYLAG6HRJBVBNBBTRTZAILU3AVCNFSM6AAAAABD6PUX2WVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAOJQGIYDCOJRGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

sergio-teruel commented:
To be taken into account: https://github.com/OCA/stock-logistics-barcode/pull/604

sergio-teruel commented:
> @chienandalu I'm having a real trouble figuring how the code works.
> 
> When you use the barcode button on a picking type. It calls `action_barcode_scan` on the `stock_picking_type` models there is 3 methods that are called:
> 
> * `create`
> * `determine_todo_action`
> * `fill_pending_moves`
> 
> 1. First called is `create` this call `set_candidate_picking` if there is a `picking_id` but there is no `picking_id` in the parameters (even when we click on the new button on a picking type). It then calls `action_show_step` which sets the correct step because most `filled_default` are still correctly set (by `_prepare_barcode_wiz_vals`)
> 2. Second is `determine_todo_action`, if the barcode option is guided, this calls `fill_todo_records` if there is no `todo_line_ids` (and there isn't) which calls `_get_stock_move_lines`. Since there is no `picking_id`, `_get_stock_move_lines` has to return an empty list. In this method all the default are filled but since there is no `todo_line_ids` all those value are set to empty sets. This then calls `action_show_step` again which remove the properly set step because the default value are now all empty.
> 3. Third a call to `fill_pending_moves` this call `fill_todo_records` again which still does nothing.
> 
> So during `action_barcode_scan`:
> 
> * `create` sets the proper default values
> * `action_show_step` is called twice once it sets a step (during) `create` and the second time it undoes what it did because `determine_todo_action` unset all the default values
> * `fill_todo_records` is called twice and twice it does nothing
> 
> This means no step at the end in some case (which are real convoluted but not so much).
> 
> * Set the barcode option as guided
> * Leave no step with a step number of 0 (if a step with a 0 selected is `action_show_step` it will trigger the line`self.step = options_required[:1].step`because `not 0` is `True`).
> * And click on the barcode of the `picking_type`
> 
> Or simpler use the demo Picking internals barcode option on internal transfer and click on the barcode on the internal transfer picking type.

The next week in Spanish OCA days we will work about it...

pedrobaeza commented:
Superseded by #610, that was the result of the Spanish OCA days,
